### PR TITLE
Adding probes for sampling

### PR DESCRIPTION
**Pull request by gantech**

### CODE REVIEW

File: Pedersen_N02/AmrWind/abl.inp [hunk 1]
```diff
@@ -77,3 +77,81 @@ diffusion.mg_rtol = 1.0e-6
...
+sampling.output_frequency = 5
+# Type of probes to output
+sampling.labels = l_v11 l_v12 l_v13 l_v21 l_v22 l_v23 l_v31 l_v32 l_v33 p_h p_v1 p_v2
+
+# Fields to output
+sampling.fields = velocity temperature
+
+# Definitions for each probe
+sampling/l_v11.type = LineSampler
+sampling/l_v11.num_points = 192
```
Comment (sayerhs) [inline Pedersen_N02/AmrWind/abl.inp:94]:
```suggestion
sampling/l_v11.num_points = 201
```

File: Pedersen_N02/AmrWind/abl.inp [hunk 2]
```diff
@@ -77,3 +77,81 @@ diffusion.mg_rtol = 1.0e-6
...
+sampling/l_v11.start = 500.0 500.0 5.208333333333333
+sampling/l_v11.end = 500.0 500.0 1994.7916666666667
+
+sampling/l_v12.type = LineSampler
+sampling/l_v12.num_points = 192
```
Comment (sayerhs) [inline Pedersen_N02/AmrWind/abl.inp:99]:
```suggestion
sampling/l_v12.num_points = 201
```

File: Pedersen_N02/AmrWind/abl.inp [hunk 3]
```diff
@@ -77,3 +77,81 @@ diffusion.mg_rtol = 1.0e-6
...
+sampling/l_v12.start = 500.0 1500.0 5.208333333333333
+sampling/l_v12.end = 500.0 1500.0 1994.7916666666667
+
+sampling/l_v13.type = LineSampler
+sampling/l_v13.num_points = 192
```
Comment (sayerhs) [inline Pedersen_N02/AmrWind/abl.inp:104]:
```suggestion
sampling/l_v13.num_points = 201
```

File: Pedersen_N02/AmrWind/abl.inp [hunk 4]
```diff
@@ -77,3 +77,81 @@ diffusion.mg_rtol = 1.0e-6
...
+sampling/l_v13.start = 500.0 2500.0 5.208333333333333
+sampling/l_v13.end = 500.0 2500.0 1994.7916666666667
+
+sampling/l_v21.type = LineSampler
+sampling/l_v21.num_points = 192
```
Comment (sayerhs) [inline Pedersen_N02/AmrWind/abl.inp:109]:
```suggestion
sampling/l_v21.num_points = 201
```

File: Pedersen_N02/AmrWind/abl.inp [hunk 5]
```diff
@@ -77,3 +77,81 @@ diffusion.mg_rtol = 1.0e-6
...
+sampling/l_v21.start = 1500.0 500.0 5.208333333333333
+sampling/l_v21.end = 1500.0 500.0 1994.7916666666667
+
+sampling/l_v22.type = LineSampler
+sampling/l_v22.num_points = 192
```
Comment (sayerhs) [inline Pedersen_N02/AmrWind/abl.inp:114]:
```suggestion
sampling/l_v22.num_points = 201
```

File: Pedersen_N02/AmrWind/abl.inp [hunk 6]
```diff
@@ -77,3 +77,81 @@ diffusion.mg_rtol = 1.0e-6
...
+sampling.labels = l_v11 l_v12 l_v13 l_v21 l_v22 l_v23 l_v31 l_v32 l_v33 p_h p_v1 p_v2
+
+# Fields to output
+sampling.fields = velocity temperature
+
+# Definitions for each probe
+sampling/l_v11.type = LineSampler
+sampling/l_v11.num_points = 192
+sampling/l_v11.start = 500.0 500.0 5.208333333333333
+sampling/l_v11.end = 500.0 500.0 1994.7916666666667
```
Comment (sayerhs) [inline Pedersen_N02/AmrWind/abl.inp:96]:
```suggestion
sampling/l_v11.start = 500.0 500.0 0.0
sampling/l_v11.end = 500.0 500.0 2000.0
```

File: Pedersen_N02/AmrWind/abl.inp [hunk 7]
```diff
@@ -77,3 +77,81 @@ diffusion.mg_rtol = 1.0e-6
 diffusion.mg_atol = 1.0e-12
 temperature_diffusion.mg_rtol = 1.0e-10
 temperature_diffusion.mg_atol = 1.0e-13
+
+# Activate data probe sampling
+incflo.post_processing = sampling
+
+# Frequency of output for the data
+sampling.output_frequency = 5
```
Comment (lawrenceccheung) [inline Pedersen_N02/AmrWind/abl.inp:85]: Don't forget to change sampling to every iteration
```
sampling.output_frequency = 1
```

**Review by lawrenceccheung — COMMENTED** [review]
Looks good, after changes from @sayerhs it should be consistent with Nalu-Wind probes.

File: Pedersen_N02/AmrWind/abl.inp (hunk 1 above)
Comment (gantech) [reply Pedersen_N02/AmrWind/abl.inp:94]: We want the same number of points as the vertical resolution I believe. For e.g. https://github.com/Exawind/wind-energy/blob/master/Pedersen_N02/NaluWindRun01/NaluWind_N02_velocity.dat has only 193 rows.

File: Pedersen_N02/AmrWind/abl.inp (hunk 6 above)
Comment (gantech) [reply Pedersen_N02/AmrWind/abl.inp:96]: I was planning to probe only at the cell centers. Will the previous one be ok then?

File: Pedersen_N02/AmrWind/abl.inp (hunk 7 above)
Comment (gantech) [reply Pedersen_N02/AmrWind/abl.inp:85]: Thanks. I ran out an input file with a sampling frequency of 100 upto 80000s and then bumped the sampling frequency to 1. How do we want to handle this in the repo? Do we want to upload both input files?

Comment (lawrenceccheung) [reply Pedersen_N02/AmrWind/abl.inp:85]: Great, thanks @gantech.  So far I've just been uploading the initial input file, with notes in the file to "uncomment this section" when sampling for spectra.  If you want, we can upload both input files, so there's no ambiguity about how things were sampled.

File: Pedersen_N02/AmrWind/abl.inp (hunk 1 above)
Comment (gantech) [reply Pedersen_N02/AmrWind/abl.inp:94]: @lawrenceccheung Could you confirm this. I believe you're sampling at all the nodes in Nalu-wind. Hence the 193 rows right? So amr-wind should have only 192?

File: Pedersen_N02/AmrWind/abl.inp (hunk 7 above)
Comment (gantech) [reply Pedersen_N02/AmrWind/abl.inp:85]: Thanks. I'll follow the same logic then.

File: Pedersen_N02/AmrWind/abl.inp (hunk 1 above)
Comment (lawrenceccheung) [reply Pedersen_N02/AmrWind/abl.inp:94]: @gantech Oh, I should clarify, the output of the horizontally averaged files (like [NaluWind_N02_velocity.dat](https://github.com/Exawind/wind-energy/blob/master/Pedersen_N02/NaluWindRun01/NaluWind_N02_velocity.dat)) ultimately comes from the `boundary_layer_statistics` algorithm in Nalu-Wind, and not from the `data_probes` we put in for sampling spectra.  

For the sampling of spectra in `data_probes`, Alan and I  _oversampled_ the domain on purpose (which is why we put 201 points for 192 cells in height).  That way we still have at least one sample per cell, we avoid any possible issues with data points hitting the exact cell boundaries, and the heights come out in nice, round 10m increments.

**Pull request closed without merging**